contracts/test/TestStrategy.sol

```diff
@@ -24,67 +24,51 @@ contract TestStrategy is BaseStrategy {
-        return (100 * gasCost < credit.add(profit));
-    }
-
-    function expectedReturn() public override view returns (uint256) {
-        return vault.expectedReturn();
-    }
-
     function estimatedTotalAssets() public override view returns (uint256) {
+        // For mock, this is just everything we have
         return want.balanceOf(address(this));
     }
 
-    function prepareReturn() internal override {
+    function prepareReturn(uint256 _debtOutstanding) internal override returns (uint256 _profit) {
         // During testing, send this contract some tokens to simulate "Rewards"
+        uint256 reserve = getReserve();
+        uint256 total = want.balanceOf(address(this));
+        if (total > reserve.add(_debtOutstanding)) _profit = total.sub(reserve).sub(_debtOutstanding);
     }
 
-    function adjustPosition() internal override {
+    function adjustPosition(uint256 _debtOutstanding) internal override {
         // Whatever we have "free", consider it "invested" now
-        if (outstanding <= want.balanceOf(address(this))) {
-            reserve = want.balanceOf(address(this)).sub(outstanding);
+        uint256 total = want.balanceOf(address(this));
+        if (total > _debtOutstanding) {
+            setReserve(total.sub(_debtOutstanding));
         } else {
-            reserve = 0;
+            setReserve(0);
         }
     }
 
-    function liquidatePosition(uint256 _amount) internal override {
-        if (_amount > reserve) {
-            reserve = 0;
+    function liquidatePosition(uint256 _amountNeeded) internal override returns (uint256 _amountFreed) {
+        uint256 reserve = getReserve();
+        if (_amountNeeded >= reserve) {
+            // Give back the entire reserves
+            _amountFreed = reserve;
         } else {
-            reserve = reserve.sub(_amount);
+            // Give back a portion of the reserves
+            _amountFreed = _amountNeeded;
         }
+        setReserve(reserve.sub(_amountFreed));
     }
 
     function exitPosition() internal override {
-        // Dump 25% each time this is called, the first 3 times
-        reserve = want.balanceOf(address(this)).mul(countdownTimer).div(4);
-        countdownTimer.sub(1); // NOTE: This should never be called after it hits 0
+        // Dump 1/N of original position each time this is called
+        setReserve(want.balanceOf(address(this)).mul(countdownTimer.sub(1)).div(countdownTimer));
+        countdownTimer = countdownTimer.sub(1); // NOTE: This should never be called after it hits 0
     }
 
     function prepareMigration(address _newStrategy) internal override {
-        want.transfer(_newStrategy, want.balanceOf(address(this)));
+        // Nothing needed here because no additional tokens/tokenized positions for mock
+    }
+
+    function protectedTokens() internal override view returns (address[] memory) {
+        return new address[](0); // No additional tokens/tokenized positions for mock
```

Comment:
I think an example should include at least one token
```suggestion
        address[] memory protected = new address[](1);
        protected[0] = 0x00000000000000000000000000000000DeaDBeef;
        return protected;
```

Reply:
`BaseStrategy` has an example already. `TestStrategy` isn't really meant to be used as an example.